### DISCUSSION TITLE
(afl++) PROGRAM ABORT : All test cases time out or crash, giving up!

### DISCUSSION THREAD

**Discussion by Janette88** [AFLplusplus/AFLplusplus#1455] [Q&A]
when i used afl++ fuzz gimp, i got the error like this:
run the command in terminal:
$ ASAN_OPTIONS=detect_leaks=0,abort_on_error=1,symbolize=0 afl-fuzz -i './afl_in' -o './afl_out' -D -t 100 -- ./install/bin/gimp-console-2.8 --verbose -d -f @@

afl-fuzz++4.01a based on afl by Michal Zalewski and a large online community
[+] afl++ is maintained by Marc "van Hauser" Heuse, Heiko "hexcoder" Ei?feldt, Andrea Fioraldi and Dominik Maier
[+] afl++ is open source, get it at https://github.com/AFLplusplus/AFLplusplus
[+] NOTE: This is v3.x which changes defaults and behaviours - see README.md
[+] Loaded environment variable ASAN_OPTIONS with value detect_leaks=0,abort_on_error=1,symbolize=0
[+] No -M/-S set, autoconfiguring for "-S default"
[*] Getting to work...
[+] Using exponential power schedule (FAST)
[+] Enabled testcache with 50 MB
[+] Generating fuzz data with a length of min=1 max=1048576
[*] Checking core_pattern...
[!] WARNING: Could not check CPU scaling governor
[+] You have 1 CPU core and 1 runnable tasks (utilization: 100%).
[*] Setting up output directories...
[+] Output directory exists but deemed OK to reuse.
[*] Deleting old session data...
[+] Output dir cleanup successful.
[*] Scanning './afl_in'...
[+] Loaded a total of 5 seeds.
[*] Creating hard links for all input files...
[*] Validating target binary...
[*] Spinning up the fork server...
[+] All right - fork server is up.
[*] Target map size: 73933
[*] No auto-generated dictionary tokens to reuse.
[*] Attempting dry run with 'id:000000,time:0,execs:0,orig:tga.tga'...
[!] WARNING: Test case results in a timeout (skipping)
[*] Attempting dry run with 'id:000001,time:0,execs:0,orig:tga-rle.tga'...
[!] WARNING: Test case results in a timeout (skipping)
[*] Attempting dry run with 'id:000002,time:0,execs:0,orig:gimpbrush.gbr'...
[!] WARNING: Test case results in a timeout (skipping)
[*] Attempting dry run with 'id:000003,time:0,execs:0,orig:gimp.xcf'...
[!] WARNING: Test case results in a timeout (skipping)
[*] Attempting dry run with 'id:000004,time:0,execs:0,orig:SampleInput.xcf'...
[!] WARNING: Test case results in a timeout (skipping)

[-] PROGRAM ABORT : All test cases time out or crash, giving up!
         Location : perform_dry_run(), src/afl-fuzz-init.c:1131
how can i start the afl-fuzz normally?  what's wrong with the fuzz? For help,thanks~~~

**Comment by vanhauser-thc** [comment]
well you should sent a good -t value to afl-fuzz. your target is very slow, too slow, working on the testcases.

time  ./install/bin/gimp-console-2.8 --verbose -d -f afl_in/one_of_the_files_there

and then add 25-50% and set this in milliseconds as the -t parameter to afl-fuzz.

**Reply by Janette88** [reply]
thanks a lot @vanhauser-thc . Even though i still stuck in the  problem, the method you told me is a good way to improve testing my target.

**Reply by vanhauser-thc** [reply]
still the same problem? you can also try something like 10000+ (yes including the plus sign)
